server/aesdsocket.c

```diff
@@ -0,0 +1,253 @@
+    {
+        perror("setsockopt"); // Print an error message if setsockopt fails
+        syslog(LOG_ERR, "Error setting socket options: %m"); // Log setsockopt error
+        close(server_socket); // Close the server socket
+        return -1; // Return an error status
+    }
+
+    struct sockaddr_in server_addr;
+    memset(&server_addr, 0, sizeof(server_addr));
+    server_addr.sin_family = AF_INET;
+    server_addr.sin_port = htons(PORT);
+    server_addr.sin_addr.s_addr = INADDR_ANY; 
+
+    // Bind socket to port 9000
+    if (bind(server_socket, (struct sockaddr *)&server_addr, sizeof(server_addr)) == -1) {
+        perror("Socket bind failed"); // Print an error message if socket bind fails
+        syslog(LOG_ERR, "Socket bind failed: %m"); // Log socket bind error
+        close(server_socket); // Close the server socket
+        return -1; // Return an error status
+    }
+
+    // Listen for incoming connections
+    if (listen(server_socket, 1) == -1) {
+        perror("Listen failed"); // Print an error message if listen fails
+        syslog(LOG_ERR, "Listen failed: %m"); // Log listen error
+        close(server_socket); // Close the server socket
+        return -1; // Return an error status
+    }
+
+    while (1) 
+    {
+        struct sockaddr_in client_addr;
+        socklen_t client_len = sizeof(client_addr);
+
+        client_socket = accept(server_socket, (struct sockaddr *)&client_addr, &client_len);
+        if (client_socket == -1) {
+            perror("Accept failed"); // Print an error message if accept fails
+            syslog(LOG_ERR, "Accept failed: %m"); // Log accept error
+        }
+        else {
+            char client_ip[INET_ADDRSTRLEN];
+
+            getpeername(client_socket, (struct sockaddr *)&client_addr, &client_len);
+
+            inet_ntop(AF_INET, &(client_addr.sin_addr), client_ip, INET_ADDRSTRLEN);
+
+            // Log accepted connection
+            syslog(LOG_INFO, "Accepted connection from %s", client_ip);
+
+            char *buffer = (char *)malloc(sizeof(char) * BUFFER_SIZE);
+            if (buffer == NULL) {
+                syslog(LOG_ERR, "Memory allocation failed!"); // Log memory allocation error
+                return -1; // Return an error status
+            }
+
+            // Receive and append data
+            int data_file = open(DATA_FILE, O_WRONLY | O_CREAT | O_APPEND, FILE_PERMISSIONS);
+            if (data_file == -1) {
+                perror("File open failed"); // Print an error message if file open fails
+                syslog(LOG_ERR, "File open failed: %m"); // Log file open error
+                free(buffer);
+                close(client_socket);
+            }
+            else {
+                ssize_t bytes_received;
+                while ((bytes_received = recv(client_socket, buffer, BUFFER_SIZE, 0)) > 0) {
+                    write(data_file, buffer, bytes_received);
```

Comment:
What if \n was not the last character? you are still writing the unrequired characters to the data_file fd

Reply:
Yeah I totally missed it, will add it. Thanks for pointing it out.